### ISSUE TITLE
Ctrl+Shift+B -> cl.exe build active file -> CRASH

### ISSUE THREAD

**Issue by FriskyDev** [microsoft/vscode-cpptools#3759]
Issue Type: <b>Bug</b>

1. Create  a folder
2. Add hello world code

```
#include <stdlib.h>
int main()
{
    return EXIT_SUCCESS;
}
```

3. Type Ctrl+Shift+B
4. Select "C/C++: cl.exe build active file"
5. Watch the BOOM

Extension version: 0.23.1
VS Code version: Code 1.35.1 (c7d83e57cd18f18026a8162d042843bda1bcf21f, 2019-06-12T15:16:14.389Z)
OS version: Windows_NT ia32 10.0.18362

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz (2 x 3192)|
|GPU Status|2d_canvas: disabled_software<br>checker_imaging: disabled_off<br>flash_3d: disabled_software<br>flash_stage3d: disabled_software<br>flash_stage3d_baseline: disabled_software<br>gpu_compositing: disabled_software<br>multiple_raster_threads: disabled_off<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>surface_synchronization: disabled_off<br>video_decode: disabled_software<br>webgl: disabled_off<br>webgl2: disabled_off|
|Load (avg)|undefined|
|Memory (System)|1.60GB (0.56GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>

**Comment by FriskyDev** [comment]
![image](https://user-images.githubusercontent.com/1686727/59389806-2c1dcb80-8d24-11e9-82ac-c53725d5fd09.png)

**Comment by FriskyDev** [comment]
So, apparently this is caused by opening the Terminal window. Right after I filed this, I reloaded so that I could cl.exe from the command line, and when I `` Ctrl+` ``, or `View -> Terminal Window`, I get the same crash as above.

Note, this is a brand new fresh install of 19H1, followed by a fresh install of VS Code, followed by a fresh install of Visual Studio 2019 Community (C++ parts only, of course ;).

**Comment by sean-mcmanus** [comment]
For cl.exe, you need to launch Code from the VS developer command prompt or you should get an error message from us.

I'm not able to repro the crash, but it sounds like a VS Code bug. Our extension normally should not be able to cause a "The window has crashed" result.

**Comment by FriskyDev** [comment]
Agreed. It looks like opening the terminal window is actually the culprit. I suspect the first thing that happens is that a terminal windows is opened. Doing this by itself crashes VS Code in the same way.

In running this repro several times, at one point I did see the message that you must be running from a command prompt (which is normally how I roll). When I opened a developer prompt and opened the folder with `code .` from the command prompt, I got the same results again. Likely the terminal window issue.

Closing...
